linera-storage/src/lib.rs

```diff
@@ -470,7 +470,7 @@ where
         }
     }
 
-    async fn get_blob(&self, blob_id: BlobId) -> Result<BlobContent, ExecutionError> {
+    async fn get_blob_content(&self, blob_id: BlobId) -> Result<BlobContent, ExecutionError> {
         Ok(self.storage.read_blob(blob_id).await?.into_inner())
```

Comment:
I'd call it `get_blob` and return a `Blob`: To my mind, it should be the storage implementation itself that guarantees that it doesn't store wrong contents with a given ID. So it should also be storage that uses that invariant, and combines them again.